src/test/scala/org/ergoplatform/nodeView/ErgoNodeViewHolderSpecification.scala

```diff
@@ -355,6 +359,102 @@ class ErgoNodeViewHolderSpecification extends ErgoNodeViewHolderTestHelpers {
     }
   }
 
+  private val t12 = TestCase("Do not apply txs with wrong header id") { fixture =>
```

Comment:
How does this relates to current PR?

Reply:
Was in issue description. Add genesis id and some test cases.

![image](https://user-images.githubusercontent.com/1447374/44297724-590d7780-a2e7-11e8-8542-89cc0cdb7657.png)